notmuch/mutt_notmuch.c

```diff
@@ -1883,6 +1883,34 @@ int nm_read_entire_thread(struct Context *ctx, struct Header *h)
   return rc;
 }
 
+/**
+ * Parse a query type out of a query.
+ *
+ * If a user writes a query for a vfolder and includes a type= statement, that type= will
+ * be encoded, which Notmuch will treat as part of the query= statement. This method will
+ * remove the type= and set it within the Mailbox struct.
+ *
+ * @param ctx   Mailbox, used for the query_type
+ * @param buf   Buffer for URI
+ */
+void parse_nm_query_type_from_query(struct NmCtxData *data, char *buf) {
+  // The six variations of how type= could appear.
+  const char *variants[6] = {"&type=threads", "&type=messages",
+    "type=threads&", "type=messages&",
+    "type=threads", "type=messages"};
```

Comment:
This seems a bit awkward...
The first two are only correct if at the end of a query.
The second two, only at the start.
The final two only if they are the entire query.

Otherwise, respectively...

- `notmuch://?query=hello&type=threadsJUNK`
- `notmuch://?JUNKtype=threads&query=hello`
- `notmuch://?JUNKtype=threadsJUNK`

Reply:
ok, thanks